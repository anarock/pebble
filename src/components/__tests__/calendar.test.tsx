import * as React from "react";
import Calendar from "../Calendar";
import sinon from "sinon";
import { mount } from "enzyme";
import { format, getTime, startOfDay, endOfDay } from "date-fns";

const date = [new Date(2012, 11, 1), new Date(2012, 12, 1)];

describe("Calendar", () => {
  test("should return correct range value on apply", () => {
    const applySpy = sinon.spy();
    const clearSpy = sinon.spy();
    const changeSpy = sinon.spy();

    const calendar = mount(
      <Calendar
        className="calendar-test"
        onApply={applySpy}
        range
        onClear={clearSpy}
        onChange={changeSpy}
        selected={date}
      />
    );
    calendar
      .find(".react-calendar__tile")
      .at(0)
      .simulate("click");

    // in case of range selector onChange should only be called once
    // both values of range have been selected.
    expect(changeSpy.calledOnce).toBeFalsy();

    calendar
      .find(".react-calendar__tile")
      .at(10)
      .simulate("click");

    calendar
      .find(".calendar-test > div")
      .at(1)
      .find("button")
      .at(1)
      .simulate("click");

    const argument = applySpy
      .getCall(0)
      .args[0].map((x: Date) => format(x, "DD-MM-YYYY"));
    expect(argument).toEqual(["01-12-2012", "11-12-2012"]);

    calendar
      .find(".calendar-test > div")
      .at(1)
      .find("button")
      .at(0)
      .simulate("click");

    expect(clearSpy.calledOnce).toBeTruthy();
  });

  test("should return correct date value on single select after apply", () => {
    const applySpy = sinon.spy();
    const clearSpy = sinon.spy();
    const changeSpy = sinon.spy();

    const calendar = mount(
      <Calendar
        className="calendar-test"
        onApply={applySpy}
        range
        onClear={clearSpy}
        onChange={changeSpy}
        selected={date}
      />
    );

    calendar
      .find(".react-calendar__tile")
      .at(0)
      .simulate("click");

    calendar
      .find(".calendar-test > div")
      .at(1)
      .find("button")
      .at(1)
      .simulate("click");

    const argument = applySpy.getCall(0).args[0].map((x: Date) => getTime(x));
    expect(argument).toEqual([
      getTime(startOfDay("12/01/2012")),
      getTime(endOfDay("12/01/2012"))
    ]); // MM/DD/YYYY
  });

  test("should return correct value on apply", () => {
    const applySpy = sinon.spy();
    const clearSpy = sinon.spy();
    const changeSpy = sinon.spy();

    const calendar = mount(
      <Calendar
        className="calendar-test"
        onApply={applySpy}
        onClear={clearSpy}
        onChange={changeSpy}
        selected={date[0]}
      />
    );
    calendar
      .find(".react-calendar__tile")
      .at(0)
      .simulate("click");

    expect(changeSpy.calledOnce).toBeTruthy();

    calendar
      .find(".calendar-test > div")
      .at(1)
      .find("button")
      .at(1)
      .simulate("click");

    const argument = format(applySpy.getCall(0).args[0], "DD-MM-YYYY");
    expect(argument).toEqual("01-12-2012");

    calendar
      .find(".calendar-test > div")
      .at(1)
      .find("button")
      .at(0)
      .simulate("click");

    expect(clearSpy.calledOnce).toBeTruthy();
  });
});
