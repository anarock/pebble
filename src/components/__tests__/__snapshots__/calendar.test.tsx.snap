// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar snapshot 1`] = `
.emotion-39 {
  background-color: #ffffff;
  border-radius: 3px;
  box-shadow: 0 2px 15px 0 rgba(0,0,0,0.1);
  overflow: hidden;
  padding: 20px;
  position: relative;
}

.emotion-0 {
  font-weight: normal;
  color: #101721;
  font-size: 14px;
  height: 48px;
  width: 48px;
  position: relative;
  text-align: center;
  cursor: pointer;
  outline: none;
  background-color: #F7F7F9;
  border-radius: 3px;
  display: table;
  border-collapse: collapse;
  border: 1px solid #ffffff;
  border-left: 0;
  border-top: 0;
}

.emotion-0 time {
  font-family: inherit;
}

.emotion-0.react-calendar__tile--now {
  color: #6161FF;
}

.emotion-0:hover {
  background-color: #EDEDED;
}

.emotion-0[disabled] {
  color: #9FA6B0;
}

.emotion-0[disabled]:hover {
  background-color: #F7F7F9;
  cursor: not-allowed;
}

.emotion-0.react-calendar__tile--active {
  background-color: #E8E8FF;
  color: #6161FF;
}

.emotion-0.react-calendar__tile--rangeStart {
  background-color: #6161FF;
  color: #ffffff !important;
}

.emotion-0.react-calendar__tile--rangeEnd {
  background-color: #6161FF;
  color: #ffffff !important;
}

.emotion-0.react-calendar__tile--singleSelect {
  background-color: #6161FF;
  color: #ffffff;
}

.emotion-0.react-calendar__month-view__days__day--neighboringMonth {
  color: #9FA6B0;
}

.emotion-35 {
  width: 356px;
  margin: 10px;
  position: relative;
}

.emotion-35 .react-calendar__month-view__weekdays__weekday {
  color: #6B7785;
  line-spacing: 7px;
  font-size: 12px;
  text-align: center;
}

.emotion-35 .react-calendar__navigation__arrow {
  border: 1px solid #E0E0E0;
  padding: 8px 10px;
  border-radius: 3px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  outline: none;
  height: 40px;
  width: 40px;
}

.emotion-35 .react-calendar__navigation__arrow.react-calendar__navigation__prev-button {
  position: absolute;
  right: 47px;
}

.emotion-35 .react-calendar__navigation__label {
  font-weight: bold;
  color: #101721;
  font-size: 18px;
  text-align: left;
  border: 0;
  outline: none;
  cursor: pointer;
  padding-left: 5px;
}

.emotion-35 .react-calendar__month-view {
  margin-top: 30px;
}

.emotion-35 .react-calendar__month-view__weekdays {
  margin-bottom: 20px;
}

.emotion-35 .react-calendar__decade-view__years,
.emotion-35 .react-calendar__century-view,
.emotion-35 .react-calendar__year-view {
  margin-top: 15px;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  width: 100%;
}

.emotion-13 {
  height: 4px;
  width: 4px;
  display: inline-block;
  border-radius: 3px;
  margin: 5px 2px 0;
}

.emotion-38 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: initial;
  -ms-flex-line-pack: initial;
  align-content: initial;
  margin-top: 20px;
}

.emotion-36 {
  line-height: 23px;
  height: 40px;
  padding: 0 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all ease-out .2s;
  transition: all ease-out .2s;
  border-radius: 3px;
  outline: none;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  white-space: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  font-weight: normal;
  color: #101721;
  font-size: 12px;
  min-width: 100px;
  height: 40px;
  background-color: #EDEDED;
  box-shadow: none;
}

.emotion-36.__pebble__button__disabled {
  cursor: not-allowed;
}

.emotion-36.__pebble__button__disabled:focus {
  border: none;
}

.emotion-36.__pebble__button__dropdown {
  color: #101721;
  background-color: #ffffff;
}

.emotion-36.__pebble__button__dropdown:not(.__pebble__button__dropdown):not(__pebble__button__disabled):hover {
  background-color: #F7F7F9;
}

.emotion-36.__pebble__button__dropdown:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #E0E0E0;
}

.emotion-36.__pebble__button__dropdown.__pebble__button__disabled {
  color: #9FA6B0;
  background-color: #EDEDED;
}

.emotion-36.__pebble__button__dropdown__open {
  color: #ffffff;
  background-color: #6161FF;
}

.emotion-36.__pebble__button__dropdown__open:not(.__pebble__button__dropdown):not(.__pebble__button__disabled):hover {
  background-color: #7070FF;
}

.emotion-36.__pebble__button__dropdown__open:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #5656ED;
}

.emotion-36.__pebble__button__dropdown__open.__pebble__button__disabled {
  color: #ffffff;
  background-color: #C0C0FF;
}

.emotion-36.__pebble__button__dropdown__open:focus {
  border: none;
}

.emotion-36.__pebble__button__dropdown__selected {
  background-color: #6161FF;
  color: #ffffff;
}

.emotion-36:focus {
  border: none;
}

.emotion-36.__pebble__button_link {
  background-color: transparent;
  border: 0;
  color: #6161FF;
  min-width: 0;
  font-size: 14px;
}

.emotion-36.__pebble__button_link:not(.__pebble__button__disabled):hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-36.__pebble__button_link.__pebble__button__disabled {
  color: #C0C0FF;
  cursor: not-allowed;
}

.emotion-36:not(.__pebble__button__dropdown):not(__pebble__button__disabled):hover {
  background-color: #F7F7F9;
}

.emotion-36:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #E0E0E0;
}

.emotion-36.__pebble__button__disabled {
  color: #9FA6B0;
  background-color: #EDEDED;
}

.emotion-37 {
  line-height: 23px;
  height: 40px;
  padding: 0 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all ease-out .2s;
  transition: all ease-out .2s;
  border-radius: 3px;
  outline: none;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  white-space: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  font-weight: normal;
  color: #ffffff;
  font-size: 12px;
  min-width: 100px;
  height: 40px;
  background-color: #6161FF;
  box-shadow: none;
}

.emotion-37.__pebble__button__disabled {
  cursor: not-allowed;
}

.emotion-37.__pebble__button__disabled:focus {
  border: none;
}

.emotion-37.__pebble__button__dropdown {
  color: #101721;
  background-color: #ffffff;
}

.emotion-37.__pebble__button__dropdown:not(.__pebble__button__dropdown):not(__pebble__button__disabled):hover {
  background-color: #F7F7F9;
}

.emotion-37.__pebble__button__dropdown:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #E0E0E0;
}

.emotion-37.__pebble__button__dropdown.__pebble__button__disabled {
  color: #9FA6B0;
  background-color: #EDEDED;
}

.emotion-37.__pebble__button__dropdown__open {
  color: #ffffff;
  background-color: #6161FF;
}

.emotion-37.__pebble__button__dropdown__open:not(.__pebble__button__dropdown):not(.__pebble__button__disabled):hover {
  background-color: #7070FF;
}

.emotion-37.__pebble__button__dropdown__open:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #5656ED;
}

.emotion-37.__pebble__button__dropdown__open.__pebble__button__disabled {
  color: #ffffff;
  background-color: #C0C0FF;
}

.emotion-37.__pebble__button__dropdown__open:focus {
  border: none;
}

.emotion-37.__pebble__button__dropdown__selected {
  background-color: #6161FF;
  color: #ffffff;
}

.emotion-37:focus {
  border: none;
}

.emotion-37.__pebble__button_link {
  background-color: transparent;
  border: 0;
  color: #6161FF;
  min-width: 0;
  font-size: 14px;
}

.emotion-37.__pebble__button_link:not(.__pebble__button__disabled):hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-37.__pebble__button_link.__pebble__button__disabled {
  color: #C0C0FF;
  cursor: not-allowed;
}

.emotion-37:not(.__pebble__button__dropdown):not(.__pebble__button__disabled):hover {
  background-color: #7070FF;
}

.emotion-37:not(.__pebble__button__dropdown):not(__pebble__button__disabled):active {
  background-color: #5656ED;
}

.emotion-37.__pebble__button__disabled {
  color: #ffffff;
  background-color: #C0C0FF;
}

<div
  className="emotion-39"
>
  <div
    className="react-calendar emotion-35"
    onBlur={null}
    onMouseOut={null}
  >
    <div
      className="react-calendar__navigation"
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <button
        className="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon-chevron-left"
          style={
            Object {
              "fontSize": 16,
            }
          }
        />
      </button>
      <button
        className="react-calendar__navigation__label"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "flexGrow": 1,
          }
        }
        type="button"
      >
        July 2018
      </button>
      <button
        className="react-calendar__navigation__arrow react-calendar__navigation__next-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="icon-arrow-right"
          style={
            Object {
              "fontSize": 16,
            }
          }
        />
      </button>
    </div>
    <div
      className="react-calendar__month-view "
    >
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "flexGrow": 1,
              "width": "100%",
            }
          }
        >
          <div
            className="react-calendar__month-view__weekdays"
            style={
              Object {
                "display": "flex",
                "flexDirection": undefined,
                "flexWrap": "no-wrap",
              }
            }
          >
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Mon
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Tue
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Wed
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Thu
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Fri
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Sat
            </div>
            <div
              className="react-calendar__month-view__weekdays__weekday"
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
            >
              Sun
            </div>
          </div>
          <div
            className="react-calendar__month-view__days"
            style={
              Object {
                "display": "flex",
                "flexDirection": undefined,
                "flexWrap": "wrap",
              }
            }
          >
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": "85.71428571428571%",
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-01T00:00:00.000"
              >
                1
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-02T00:00:00.000"
              >
                2
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-03T00:00:00.000"
              >
                3
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-04T00:00:00.000"
              >
                4
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-05T00:00:00.000"
              >
                5
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-06T00:00:00.000"
              >
                6
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-07T00:00:00.000"
              >
                7
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-08T00:00:00.000"
              >
                8
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-09T00:00:00.000"
              >
                9
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-10T00:00:00.000"
              >
                10
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-11T00:00:00.000"
              >
                11
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-12T00:00:00.000"
              >
                12
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-13T00:00:00.000"
              >
                13
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-14T00:00:00.000"
              >
                14
              </time>
              <div
                className="emotion-16"
              >
                <span
                  className="emotion-13"
                  style={
                    Object {
                      "backgroundColor": "#6161FF",
                    }
                  }
                />
                <span
                  className="emotion-13"
                  style={
                    Object {
                      "backgroundColor": "#10AC60",
                    }
                  }
                />
                <span
                  className="emotion-13"
                  style={
                    Object {
                      "backgroundColor": "#F3AA07",
                    }
                  }
                />
              </div>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-15T00:00:00.000"
              >
                15
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-16T00:00:00.000"
              >
                16
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-17T00:00:00.000"
              >
                17
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-18T00:00:00.000"
              >
                18
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-19T00:00:00.000"
              >
                19
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-20T00:00:00.000"
              >
                20
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-21T00:00:00.000"
              >
                21
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-22T00:00:00.000"
              >
                22
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-23T00:00:00.000"
              >
                23
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-24T00:00:00.000"
              >
                24
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-25T00:00:00.000"
              >
                25
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-26T00:00:00.000"
              >
                26
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-27T00:00:00.000"
              >
                27
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-28T00:00:00.000"
              >
                28
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-29T00:00:00.000"
              >
                29
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-30T00:00:00.000"
              >
                30
              </time>
            </button>
            <button
              className="react-calendar__tile react-calendar__month-view__days__day emotion-0"
              disabled={null}
              onClick={[Function]}
              onFocus={null}
              onMouseOver={null}
              style={
                Object {
                  "flexBasis": "14.285714285714286%",
                  "marginLeft": null,
                  "maxWidth": "14.285714285714286%",
                  "overflow": "hidden",
                }
              }
              type="button"
            >
              <time
                dateTime="2018-07-31T00:00:00.000"
              >
                31
              </time>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="emotion-38"
  >
    <button
      className="emotion-36"
      onClick={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
    >
      Clear
       
      <canvas
        height={0}
        onDragOver={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "borderRadius": "inherit",
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        width={0}
      />
    </button>
    <button
      className="emotion-37"
      onClick={[Function]}
      style={
        Object {
          "width": undefined,
        }
      }
    >
      Apply
       
      <canvas
        height={0}
        onDragOver={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        style={
          Object {
            "borderRadius": "inherit",
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        width={0}
      />
    </button>
  </div>
</div>
`;
